atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/sweep/DbKvsGetCandidateCellsForSweeping.java

```diff
@@ -18,61 +18,26 @@
 
 import java.util.Iterator;
 import java.util.List;
-import java.util.Optional;
 
-import com.google.common.collect.AbstractIterator;
 import com.google.common.collect.Iterators;
 import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweeping;
 import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweepingRequest;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
-import com.palantir.atlasdb.keyvalue.dbkvs.impl.sweep.CandidatePagingState.StartingPosition;
 
 public class DbKvsGetCandidateCellsForSweeping {
 
-    private final CellTsPairLoaderFactory cellTsPairLoaderFactory;
+    private final CellTsPairLoader cellTsPairLoaderFactory;
 
-    public DbKvsGetCandidateCellsForSweeping(CellTsPairLoaderFactory cellTsPairLoaderFactory) {
+    public DbKvsGetCandidateCellsForSweeping(CellTsPairLoader cellTsPairLoaderFactory) {
         this.cellTsPairLoaderFactory = cellTsPairLoaderFactory;
     }
 
     public Iterator<List<CandidateCellForSweeping>> getCandidateCellsForSweeping(
             TableReference tableRef,
             CandidateCellForSweepingRequest request) {
-        CellTsPairLoader loader = cellTsPairLoaderFactory.createCellTsLoader(tableRef, request);
-        CandidatePagingState state = CandidatePagingState.create(request.startRowInclusive());
-        return Iterators.filter(new PageIterator(loader, state), page -> !page.isEmpty());
+        Iterator<List<CellTsPairInfo>> cellTsIter = cellTsPairLoaderFactory.createPageIterator(tableRef, request);
+        Iterator<List<CandidateCellForSweeping>> rawIter = CandidateGroupingIterator.create(cellTsIter);
+        return Iterators.filter(rawIter, page -> !page.isEmpty());
```

Comment:
Does this Iterator do an eager page load of the kind we were trying to avoid in `OracleCellTsPageLoader.PageIterator`?